xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/internal/objects/classes/ImplicitlyAllowedValuesDBListQueryBuilderTest.java

```diff
@@ -301,12 +329,34 @@ public void buildWithParent() throws Exception
         assertQuery("select distinct doc.fullName as unfilterable0, doc.fullName, doc.title, doc.parent"
             + " from XWikiDocument as doc");
 
-        this.dbListClass.setClassname("XWiki.TagClass");
+        this.dbListClass.setClassname("XWiki.XWikiUsers");
         assertQuery("select distinct doc.fullName as unfilterable0, idProp.value, valueProp.value, parentProp.value"
             + " from XWikiDocument as doc, BaseObject as obj, StringProperty as idProp, StringProperty as valueProp,"
             + " StringProperty as parentProp where doc.fullName = obj.name and obj.className = :className and"
             + " doc.fullName <> :templateName and obj.id = idProp.id.id and idProp.id.name = :idProp and"
             + " obj.id = valueProp.id.id and valueProp.id.name = :valueProp and obj.id = parentProp.id.id and"
             + " parentProp.id.name = :parentProp");
     }
+
+    private void setPropertyType() throws Exception
```

Comment:
You made the current tests pass, but you haven't added a test for the new use case that you added support for. Maybe this method could have some parameters to control the property type (``StringProperty`` vs. say ``IntegerProperty``).